### PR TITLE
feat: Add 12.1 image and update 12.* versions

### PR DESCRIPTION
**Pull request by NavarrePratt**
- Update NCCL to 2.18.1
 - Update HPC-X to 2.15

Ref: https://github.com/coreweave/infra-pm/issues/541

TODO Before Merge:
- [x] Update readme table with git commit hash when the images build.

### CODE REVIEW

File: .github/workflows/ubuntu-20.yml [hunk 1]
```diff
@@ -18,6 +18,8 @@ jobs:
       cuda-version-major: "11.7"
       nccl-version: 2.14.3-1
       cuda-samples-version: "11.6"
+      hpcx-version: "2.14"
+      hpcx-nccl-version: "2.16"
```
[the remark this answers is not in the document]
Comment (salanki) [reply .github/workflows/ubuntu-20.yml:22]: I think we can leave it tbh. People should be using newer cuda.

**Review by salanki — APPROVED** [review]

**Review by salanki — APPROVED** [review]
Update readme and merge

**Comment by NavarrePratt** [comment]
@salanki - The PR is ready to be merged given your approval. Feel free to merge yourself when approving.

**Comment by salanki** [comment]
Thank you!

**Pull request merged**